File: LABORATORIO_8/PREG01_LAB08-2024-2/Biblioteca/Producto.cpp
//
// Created by USER on 18/06/2026.
//

#include "Producto.h"

#include <cstring>

Producto::Producto() {
    nombre=nullptr;
    precio=0;
}

Producto:: ~Producto(){
    if (nombre!=nullptr) delete nombre;

}


double Producto::get_precio() const {
    return precio;
}

void Producto::set_precio(double precio) {
    this->precio = precio;
}


void Producto::get_nombre(char *cad){
    if (nombre==nullptr) cad[0] = '\0';
    else strcpy(cad, nombre);
}

void Producto::set_nombre(const char *cad){
    if (nombre!=nullptr) delete nombre;
    nombre = new char[strlen(cad)+1];
    strcpy(nombre,cad);
}

void Producto::lee(ifstream &arch) {
    char cad[100], c;
    arch.getline(cad, 100,',');
    set_nombre(cad);
    arch>>precio>>c;
}

void Producto::imprime(ofstream &arch) {
    arch<<setw(5)<<" "<<"Nombre: "<<nombre<<endl;
    arch<<setprecision(2)<<fixed<<setw(5)<<" "<<"Precio: "<<precio<<endl;
}

File: Restaurante_2024-2/src/Producto.java
import java.util.Scanner;

public abstract class Producto {
    private String nombre;
    private double precio;

    public Producto(){
        precio=0;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    //142 B Cafe 2 grande
    //142 B Agua_Mineral 2.5 pequeno
    //142 E Quesadillas 5.5 sin_picante

    public abstract String tiposoy();

    public void lee(Scanner arch){
        nombre = arch.next();
        precio = arch.nextDouble();
    }

    public void imprime(){
        System.out.println("Nombre: " + nombre);
        System.out.println("Precio: " + precio);
    }
}
